src/mon/OSDMonitor.cc

```diff
@@ -942,6 +942,18 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   }
   mapping_job.reset();
 
+  // ensure we don't have blank new_state updates.  these are interrpeted as
+  // CEPH_OSD_UP (and almost certainly not what we want!).
```

Comment:
i think, if `new_state` is `0`, then it has no effect on the new osdmap after it is applied to the old one.